core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/status/AmbariClusterStatusUpdater.java

```diff
@@ -94,12 +94,17 @@ private boolean isUpdateEnabled(ClusterStatus clusterStatus) {
 
     private boolean updateClusterStatus(Long stackId, Cluster cluster, Status newClusterStatus) {
         boolean result = false;
-        if (cluster.getStatus() != newClusterStatus) {
-            LOGGER.info("Cluster {} status is updated from {} to {}", cluster.getId(), cluster.getStatus(), newClusterStatus);
-            clusterService.updateClusterStatusByStackId(stackId, newClusterStatus);
-            result = true;
+
+        Status status = cluster.getStatus();
+        if (status != newClusterStatus) {
+            if (!status.equals(Status.MAINTENANCE_MODE_ENABLED)
+                    || (status.equals(Status.MAINTENANCE_MODE_ENABLED) && !newClusterStatus.equals(Status.AVAILABLE))) {
```

Comment:
`status.equals(Status.MAINTENANCE_MODE_ENABLED)` is always true,
`if (!status.equals(Status.MAINTENANCE_MODE_ENABLED) || !newClusterStatus.equals(Status.AVAILABLE)) {` does the same